### DISCUSSION TITLE
Supporting typescript

### DISCUSSION THREAD

**Discussion by kurtaking** [octoherd/octoherd#43] [General]
Hey, have the maintainers considered supporting typescript? I checked for existing scripts that may have added types but wasn't able to find anything.

**Comment by gr2m** [comment]
Are you asking to support running `script.ts` files out of the box with the @octoherd CLI? If so, that is not something I'd like to do, but we are considering moving to Deno for the CLI which I think supports that out of the box

**Reply by kurtaking** [reply]
Yeah, something along those lines. I want to leverage typescript when writing octoherd scripts, and I have been exploring how to set that up in its current state.

**Comment by kurtaking** [comment]
Hey @gr2m, what is the lift to move the CLI to Deno? If you'd care to go more in-depth, I'm curious to learn and find out if there is something I could get involved with.

**Reply by gr2m** [reply]
I think the next step would be to build the CLI for deno in a separate repository and then see if it can run the existing scripts. 

For node.js, we add an executable directly to each script like here:
https://github.com/octoherd/script-hello-world/blob/dd5c0cfd6ebf93ff3348f48fc1f53329c646a1f5/package.json#L7

For Deno, that shouldn't be necessary, you'd have a single CLI that you would then point to a script, e.g. http://esm.sh/@octoherd/script-hello-world, in order to run it. Deno's restrictive system access should take care to make running the scripts much more safe than running them with Node
